ideascube/serveradmin/models.py

```diff
@@ -0,0 +1,55 @@
+from django.conf import settings
+from django.db import models
+
+from ideascube.models import JSONField
+
+
+class Setting(models.Model):
+    class Meta:
+        ordering = ['-date']
+
+    namespace = models.CharField(max_length=40)
+    key = models.CharField(max_length=40)
+    value = JSONField()
+    actor = models.ForeignKey(settings.AUTH_USER_MODEL)
+    date = models.DateTimeField(auto_now=True)
+
+    @classmethod
+    def _get_typed(cls, namespace, key, type, default=None):
+        try:
+            setting = Setting.objects.get(namespace=namespace, key=key)
+
+        except Setting.DoesNotExist:
+            if default is not None:
+                return default
+
+            raise
+
+        else:
+            if isinstance(setting.value, type):
+                return setting.value
+
+            if default is not None:
```

Comment:
> We could also catch the error every where we use call for in-db setting, but for what benefit? 

This, to me, is the real argument.

There is nothing we could do with that error that would be useful to the user.

At least returning the default value gives them a degraded experience (default value instead of what they expected), which is infinitely better than no experience at all, being blocked on an error message they can't, and won't understand.